lib/jxl/dec_cache.cc

```diff
@@ -28,7 +28,7 @@ Status PassesDecoderState::PreparePipeline(ImageBundle* decoded,
                                            PipelineOptions options) {
   const FrameHeader& frame_header = shared->frame_header;
   size_t num_c = 3 + frame_header.nonserialized_metadata->m.num_extra_channels;
-  if ((frame_header.flags & FrameHeader::kNoise) != 0) {
+  if (options.render_noise && (frame_header.flags & FrameHeader::kNoise) != 0) {
```

Comment:
Not super important but I think I would tend to swap the conditions, as it saves a check most of the time (when `options.render_noise` is true but there isn’t any).

Reply:
I think the difference between both variants is too small to justify running the CI another time :)